File: demo/src/main/java/com/midas/demo/mvp/presenter/RankPresenter.java
package com.midas.demo.mvp.presenter;

import com.midas.demo.base.BasePresenter;
import com.midas.demo.mvp.contract.RankContract;
import com.midas.demo.mvp.model.RankModel;
import com.midas.demo.mvp.model.bean.HomeBean;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class RankPresenter extends BasePresenter<RankContract.View> implements RankContract.Presenter {

    private RankModel model = new RankModel();

    @Override
    public void getRankList(String url) {
        checkViewAttached();
        mViewRef.get().showLoading();
        Disposable disposable = model.getRankList(url)
                .subscribe(new Consumer<HomeBean.Issue>() {
                    @Override
                    public void accept(HomeBean.Issue issue) throws Exception {
                        mViewRef.get().setRankList(issue.getItemList());

                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        mViewRef.get().setError("error");

                    }
                });
        addSubscription(disposable);

    }
}

File: app/src/main/java/com/midas/eye/mvp/presenter/RankPresenter.kt
package com.midas.eye.mvp.presenter

import com.midas.eye.base.BasePresenter
import com.midas.eye.mvp.contract.RankContract
import com.midas.eye.mvp.model.RankModel
import com.midas.eye.net.exception.ExceptionHandle

/**
 * Created by midas on 2019/05/28.
 * desc: 获取 TabInfo Presenter
 */
class RankPresenter : BasePresenter<RankContract.View>(), RankContract.Presenter {

    private val rankModel by lazy { RankModel() }

    /**
     *  请求排行榜数据
     */
    override fun requestRankList(apiUrl: String) {
        checkViewAttached()
        mRootView?.showLoading()
        val disposable = rankModel.requestRankList(apiUrl)
                .subscribe({ issue ->
                    mRootView?.apply {
                        dismissLoading()
                        setRankList(issue.itemList)
                    }
                }, { throwable ->
                    mRootView?.apply {
                        //处理异常
                        showError(ExceptionHandle.handleException(throwable),ExceptionHandle.errorCode)
                    }
                })
        addSubscription(disposable)
    }
}